invenio_rdm_records/services/components.py

```diff
@@ -139,3 +139,56 @@ def new_version(self, identity, draft=None, record=None, **kwargs):
         # (publication date and version)
         for f in self.new_version_skip_fields:
             draft.metadata.pop(f, None)
+
+
+class ExternalPIDsComponent(ServiceComponent):
+    """."""
+
+    def _pid_exists(self, external_pid):
+        """Validate external PID with provider."""
+        pid_provider = external_pid["provider"]
+        providers = self.config.external_pids_providers
+        assert pid_provider in providers
```

Comment:
minor/general practice: It could be that is not there due to config change or migration, right? do we want hard failures (i.e. assert) or better raising a custom exception that could be treated and send back properly via REST to the user?